File: lib/models/product.dart
/// نموذج بيانات المنتج - Product Model
class Product {
  final String id;
  final String name;
  final String description;
  final double price;
  final double? discount; // خصم (اختياري)
  final String category; // ملابس، معدات، مكملات، إكسسوارات
  final List<String> images;
  final double rating;
  final int reviews;
  final List<String> sizes; // المقاسات المتوفرة
  final List<String> colors; // الألوان المتوفرة
  final bool inStock;
  final int stock; // الكمية المتوفرة

  const Product({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    this.discount,
    required this.category,
    required this.images,
    required this.rating,
    required this.reviews,
    required this.sizes,
    required this.colors,
    required this.inStock,
    required this.stock,
  });

  // السعر بعد الخصم
  double get finalPrice {
    if (discount != null && discount! > 0) {
      return price - (price * discount! / 100);
    }
    return price;
  }

  // نسبة الخصم
  String get discountText {
    if (discount != null && discount! > 0) {
      return '${discount!.toInt()}%';
    }
    return '';
  }

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] as String,
      name: json['name'] as String,
      description: json['description'] as String,
      price: (json['price'] as num).toDouble(),
      discount: json['discount'] != null ? (json['discount'] as num).toDouble() : null,
      category: json['category'] as String,
      images: List<String>.from(json['images'] as List),
      rating: (json['rating'] as num).toDouble(),
      reviews: json['reviews'] as int,
      sizes: List<String>.from(json['sizes'] as List),
      colors: List<String>.from(json['colors'] as List),
      inStock: json['inStock'] as bool,
      stock: json['stock'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'price': price,
      'discount': discount,
      'category': category,
      'images': images,
      'rating': rating,
      'reviews': reviews,
      'sizes': sizes,
      'colors': colors,
      'inStock': inStock,
      'stock': stock,
    };
  }
}

File: backend/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Product extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'name',
        'name_ar',
        'slug',
        'description',
        'description_ar',
        'category',
        'category_id',
        'sku',
        'barcode',
        'price',
        'sale_price',
        'cost_price',
        'stock',
        'quantity',
        'low_stock_threshold',
        'image',
        'images',
        'specifications',
        'brand',
        'weight',
        'dimensions',
        'is_active',
        'is_featured',
        'rating',
        'reviews_count',
        'sales_count',
    ];

    protected $casts = [
        'price' => 'decimal:2',
        'sale_price' => 'decimal:2',
        'cost_price' => 'decimal:2',
        'images' => 'array',
        'specifications' => 'array',
        'dimensions' => 'array',
        'is_active' => 'boolean',
        'is_featured' => 'boolean',
        'rating' => 'decimal:1',
    ];

    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    public function orderItems()
    {
        return $this->hasMany(OrderItem::class);
    }

    public function orders()
    {
        return $this->belongsToMany(Order::class, 'order_items')
            ->withPivot('quantity', 'price', 'total');
    }

    public function getStockStatus(): string
    {
        $stock = $this->stock ?? $this->quantity ?? 0;
        if ($stock <= 0) return 'out_of_stock';
        if ($stock <= ($this->low_stock_threshold ?? 10)) return 'low_stock';
        return 'in_stock';
    }

    public function getCurrentPrice(): float
    {
        return $this->sale_price && $this->sale_price < $this->price
            ? $this->sale_price
            : $this->price;
    }

    public function getDiscountPercentage(): ?int
    {
        if (!$this->sale_price || $this->sale_price >= $this->price) return null;
        return round((($this->price - $this->sale_price) / $this->price) * 100);
    }
}
